src/modules/cpu/kernel/tensor_mean.hpp

```diff
@@ -0,0 +1,862 @@
+#endif
+            mean = (Rpp32f)sum / totalPixelsPerChannel;
+            tensorMeanArr[batchCount] = mean;
+        }
+
+        // Tensor Mean without fused output-layout toggle 3 channel (NCHW)
+        else if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NCHW))
+        {
+            Rpp64u sum;
+            Rpp32u sumR = 0, sumG = 0, sumB = 0;
+            Rpp32f mean, meanR = 0.0, meanG = 0.0, meanB = 0.0;
+            Rpp32u sumAvxR[8] = {0};
+            Rpp32u sumAvxG[8] = {0};
+            Rpp32u sumAvxB[8] = {0};
+
+            Rpp8u *srcPtrRowR, *srcPtrRowG, *srcPtrRowB, *dstPtrRow;
+            srcPtrRowR = srcPtrChannel;
+            srcPtrRowG = srcPtrRowR + srcDescPtr->strides.cStride;
+            srcPtrRowB = srcPtrRowG + srcDescPtr->strides.cStride;
+#if __AVX2__
+            __m256i pSumR = _mm256_setzero_si256();
+            __m256i pSumG = _mm256_setzero_si256();
+            __m256i pSumB = _mm256_setzero_si256();
+#endif
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                Rpp8u *srcPtrTempR, *srcPtrTempG, *srcPtrTempB;
+                srcPtrTempR = srcPtrRowR;
+                srcPtrTempG = srcPtrRowG;
+                srcPtrTempB = srcPtrRowB;
+
+                int vectorLoopCount = 0;
+#if __AVX2__
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrementPerChannel)
+                {
+                    __m256i p[6];
+                    rpp_simd_load(rpp_load48_u8pln3_to_u32pln3_avx, srcPtrTempR, srcPtrTempG, srcPtrTempB, p);
+                    compute_sum_48_host(p, &pSumR, &pSumG, &pSumB);
+                    srcPtrTempR += vectorIncrementPerChannel;
+                    srcPtrTempG += vectorIncrementPerChannel;
+                    srcPtrTempB += vectorIncrementPerChannel;
+                }
+#endif
+                for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                {
+                    sumR += (Rpp32u)(*srcPtrTempR++);
+                    sumG += (Rpp32u)(*srcPtrTempG++);
+                    sumB += (Rpp32u)(*srcPtrTempB++);
+                }
+                srcPtrRowR += srcDescPtr->strides.hStride;
+                srcPtrRowG += srcDescPtr->strides.hStride;
+                srcPtrRowB += srcDescPtr->strides.hStride;
+            }
+#if __AVX2__
+            _mm256_store_si256((__m256i *)sumAvxR, pSumR);
+            _mm256_store_si256((__m256i *)sumAvxG, pSumG);
+            _mm256_store_si256((__m256i *)sumAvxB, pSumB);
+            sumR += (sumAvxR[0] + sumAvxR[1] + sumAvxR[2] + sumAvxR[3] + sumAvxR[4] + sumAvxR[5] + sumAvxR[6] + sumAvxR[7]);
+            sumG += (sumAvxG[0] + sumAvxG[1] + sumAvxG[2] + sumAvxG[3] + sumAvxG[4] + sumAvxG[5] + sumAvxG[6] + sumAvxG[7]);
+            sumB += (sumAvxB[0] + sumAvxB[1] + sumAvxB[2] + sumAvxB[3] + sumAvxB[4] + sumAvxB[5] + sumAvxB[6] + sumAvxB[7]);
```

Comment:
Have we tried using [rpp_hsum_ps](rpp_hsum_ps) for getting the final sum?
instead of storing into another buffer and adding them?